s3-sink-connector/src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -626,28 +272,11 @@ public Template getPrefixTemplate() {
     }
 
     public ZoneId getTimezone() {
-        return ZoneId.of(getString(TIMESTAMP_TIMEZONE));
+        return ZoneId.of(getString(S3ConfigFragment.TIMESTAMP_TIMEZONE));
     }
 
     public TimestampSource getTimestampSource() {
-        return TimestampSource.of(getTimezone(), TimestampSource.Type.of(getString(TIMESTAMP_SOURCE)));
-    }
-
-    public AwsStsRole getStsRole() {
-        return new AwsStsRole(getString(AWS_STS_ROLE_ARN), getString(AWS_STS_ROLE_EXTERNAL_ID),
-                getString(AWS_STS_ROLE_SESSION_NAME), getInt(AWS_STS_ROLE_SESSION_DURATION));
-    }
-
-    public boolean hasAwsStsRole() {
-        return getStsRole().isValid();
-    }
-
-    public boolean hasStsEndpointConfig() {
-        return getStsEndpointConfig().isValid();
-    }
-
-    public AwsStsEndpointConfig getStsEndpointConfig() {
-        return new AwsStsEndpointConfig(getString(AWS_STS_CONFIG_ENDPOINT), getString(AWS_S3_REGION_CONFIG));
+        return TimestampSource.of(getTimezone(), TimestampSource.Type.of(getString(S3ConfigFragment.TIMESTAMP_SOURCE)));
```

Comment:
Not at this line, but my comment is about unused protected class AwsRegionValidator which can be deleted, as its also defined in CommonConfig